home_work_container_logging/Logger.h

```diff
@@ -0,0 +1,116 @@
+//
+// Created by flyon21 on 09.11.24.
+//
+
+#ifndef C_PRO_LOGGER_H
+#define C_PRO_LOGGER_H
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <map>
+#include <mutex>
+#include <iomanip>
+#include <vector>
+#include <mutex>
+
+
+class Logger {
+public:
+    enum LogLevel {
+        INFO, WARNING, ERROR
+    };
+
+    static Logger &getInstance() {
+        static Logger instance;
+        return instance;
+    }
+
+    void log(const std::string &message, LogLevel level = INFO) {
+        std::lock_guard<std::mutex> lock(mutex_);
+        std::ofstream logFile("log.txt", std::ios::app);
+        ColorCode coloredLogLevel = getColorCode(level);
+
+        if (logFile.is_open()) {
+            logFile << "[" << getLogLevelString(level) << "] "
+                    << getCurrentTime() << " - " << message << std::endl;
+            logFile.close();
+        } else {
+            ColorCode coloredLogLevelError = getColorCode(ERROR);
+            std::cerr << coloredLogLevelError.start << "Error opening log file!" << coloredLogLevelError.end << std::endl;
+        }
+        std::cout << coloredLogLevel.start << "[" << getLogLevelString(level) << "] "
+                  << getCurrentTime() << " - " << message << coloredLogLevel.end << std::endl;
+    }
+
+
+    template <typename Container>
+    void logContainer(const Container& container, LogLevel level = INFO) {
```

Comment:
LogLevel level could be const. It will guarantee that this data will not be changed accidentally. But it's up to you